aa_modules/aamod_firstlevel_model.xml

```diff
@@ -36,7 +36,11 @@
 			otherwise (0-1) - fraction threshold
 			-->
             <firstlevelmasking>1</firstlevelmasking>
-            
+
+            <!-- optional explicit mask (fullpath to .nii file) applied to all subjects -->
+            <!-- SPM will combine (intersect) this with firstlevelmasking option -->
+            <explicitmaskfname></explicitmaskfname>
```

Comment:
Would not be more c-aa-nonical to add a mask as a stream?

Reply:
You can use study-level stream by using `aas_getfile_bystream_multilevel`

Reply:
I've looked into aas_getfiles_bystream_multilevel and I don't understand how it solves the problem. 

The examples I can find in the code appear to be modules that can run in different domains (often aliased) and all of them pass the current domain to aas_getfiles_bystream_multilevel (for example, using the variable aap.tasklist.currenttask.domain).

This is not what I need. 

I need the user to be able to specify an explicit mask at the subject level -- for example:

`aap = aas_addinitialstream(aap, 'explicitmask', 'sub-01', 'mask.nii');
`

or at the study level -- for example:

`aap = aas_addinitialstream(aap,'explicitmask','mask.nii');
`

(I suppose a user could specify the mask at the session level, but that seems unlikely)

The problem is aamod_firstlevel_model *always* runs as subject domain. 

So I don't need the domain of the _module_ (e.g., aap.tasklist.currenttask.domain), I need the domain of the _stream_ (so I can pass it to aas_getfiles_bystream_multilevel).

 Is there a function to do that? Or am I not understanding something? (Answer: probably :-)

Reply:
You can still use the _aas_getfiles_bystream_multilevel_ function with the lowest possible stream level (i.e., subject). If it cannot find the stream at that level, it keeps looking at one level up.

It is a slightly hacky workaround, I am afraid, and I actually implemented an explicit stream level independent from the task level in reproa.